In [33]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [34]:
file =  pd.read_csv("netflix-rotten-tomatoes-metacritic-imdb.csv")
file.head()

Title                                   Genre  \
0     Lets Fight Ghost  Crime, Drama, Fantasy, Horror, Romance   
1  HOW TO BUILD A GIRL                                  Comedy   
2           Centigrade                         Drama, Thriller   
3                ANNE+                                   Drama   
4                Moxie                 Animation, Short, Drama   

                                                Tags         Languages  \
0  Comedy Programmes,Romantic TV Comedies,Horror ...  Swedish, Spanish   
1       Dramas,Comedies,Films Based on Books,British           English   
2                                          Thrillers           English   
3        TV Dramas,Romantic TV Dramas,Dutch TV Shows           Turkish   
4  Social Issue Dramas,Teen Movies,Dramas,Comedie...           English   

  Series or Movie  Hidden Gem Score  \
0          Series               4.3   
1           Movie               7.0   
2           Movie               6.4   
3          Series               7.7   
4           Movie               8.1   

                                Country Availability       Runtime  \
0                                           Thailand  < 30 minutes   
1                                             Canada      1-2 hour   
2                                             Canada      1-2 hour   
3                                Belgium,Netherlands  < 30 minutes   
4  Lithuania,Poland,France,Iceland,Italy,Spain,Gr...      1-2 hour   

          Director                      Writer  ... Netflix Release Date  \
0  Tomas Alfredson       John Ajvide Lindqvist  ...           2021-03-04   
1    Coky Giedroyc               Caitlin Moran  ...           2021-03-04   
2    Brendan Walsh  Brendan Walsh, Daley Nixon  ...           2021-03-04   
3              NaN                         NaN  ...           2021-03-04   
4    Stephen Irwin                         NaN  ...           2021-03-04   

                         Production House  \
0               Canal+, Sandrew Metronome   
1  Film 4, Monumental Pictures, Lionsgate   
2                                     NaN   
3                                     NaN   
4                                     NaN   

                             Netflix Link  \
0  https://www.netflix.com/watch/81415947   
1  https://www.netflix.com/watch/81041267   
2  https://www.netflix.com/watch/81305978   
3  https://www.netflix.com/watch/81336456   
4  https://www.netflix.com/watch/81078393   

                              IMDb Link  \
0  https://www.imdb.com/title/tt1139797   
1  https://www.imdb.com/title/tt4193072   
2  https://www.imdb.com/title/tt8945942   
3  https://www.imdb.com/title/tt6132758   
4  https://www.imdb.com/title/tt2023611   

                                             Summary  IMDb Votes  \
0  A med student with a supernatural gift tries t...    205926.0   
1  When nerdy Johanna moves to London, things get...      2838.0   
2  Trapped in a frozen car during a blizzard, a p...      1720.0   
3  Upon moving into a new place, a 20-something r...      1147.0   
4  Inspired by her moms rebellious past and a con...        63.0   

                                               Image  \
0  https://occ-0-4708-64.1.nflxso.net/dnm/api/v6/...   
1  https://occ-0-1081-999.1.nflxso.net/dnm/api/v6...   
2  https://occ-0-1081-999.1.nflxso.net/dnm/api/v6...   
3  https://occ-0-1489-1490.1.nflxso.net/dnm/api/v...   
4  https://occ-0-4039-1500.1.nflxso.net/dnm/api/v...   

                                              Poster  \
0  https://m.media-amazon.com/images/M/MV5BOWM4NT...   
1  https://m.media-amazon.com/images/M/MV5BZGUyN2...   
2  https://m.media-amazon.com/images/M/MV5BODM2MD...   
3  https://m.media-amazon.com/images/M/MV5BNWRkMz...   
4  https://m.media-amazon.com/images/M/MV5BODYyNW...   

                                  TMDb Trailer Trailer Site  
0                                          NaN          NaN  
1  https://www.youtube.com/watch?v=eIbcxPy4okQ      Yo

In [35]:
file.describe()

Hidden Gem Score    IMDb Score  Rotten Tomatoes Score  \
count      13379.000000  13381.000000            6382.000000   
mean           5.937551      6.496054              59.523034   
std            2.250202      1.146910              26.999173   
min            0.600000      1.000000               0.000000   
25%            3.800000      5.800000              38.000000   
50%            6.800000      6.600000              64.000000   
75%            7.900000      7.300000              83.000000   
max            9.800000      9.700000             100.000000   

       Metacritic Score  Awards Received  Awards Nominated For    IMDb Votes  
count       4336.000000      6075.000000           7661.000000  1.337900e+04  
mean          56.813653         8.764444             13.983161  4.272841e+04  
std           17.582545        18.311171             29.821052  1.257012e+05  
min            5.000000         1.000000              1.000000  5.000000e+00  
25%           44.000000         1.000000              2.000000  4.035000e+02  
50%           57.000000         3.000000              5.000000  2.322000e+03  
75%           70.000000         8.000000             12.000000  2.089050e+04  
max          100.000000       300.000000            386.000000  2.354197e+06

In [36]:
file.describe

<bound method NDFrame.describe of                                            Title  \
0                               Lets Fight Ghost   
1                            HOW TO BUILD A GIRL   
2                                     Centigrade   
3                                          ANNE+   
4                                          Moxie   
...                                          ...   
15475                     K-POP Extreme Survival   
15476            DreamWorks Shreks Swamp Stories   
15477  DreamWorks Happy Holidays from Madagascar   
15478                DreamWorks Holiday Classics   
15479   DreamWorks Kung Fu Panda Awesome Secrets   

                                        Genre  \
0      Crime, Drama, Fantasy, Horror, Romance   
1                                      Comedy   
2                             Drama, Thriller   
3                                       Drama   
4                     Animation, Short, Drama   
...                                       ...  

Tere are total number of 15479 rows and 29 columns

## Checking Data types and Column names

In [37]:
file.dtypes

Title                     object
Genre                     object
Tags                      object
Languages                 object
Series or Movie           object
Hidden Gem Score         float64
Country Availability      object
Runtime                   object
Director                  object
Writer                    object
Actors                    object
View Rating               object
IMDb Score               float64
Rotten Tomatoes Score    float64
Metacritic Score         float64
Awards Received          float64
Awards Nominated For     float64
Boxoffice                 object
Release Date              object
Netflix Release Date      object
Production House          object
Netflix Link              object
IMDb Link                 object
Summary                   object
IMDb Votes               float64
Image                     object
Poster                    object
TMDb Trailer              object
Trailer Site              object
dtype: object

In [38]:
file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15480 entries, 0 to 15479
Data columns (total 29 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Title                  15480 non-null  object 
 1   Genre                  13770 non-null  object 
 2   Tags                   15413 non-null  object 
 3   Languages              13545 non-null  object 
 4   Series or Movie        15480 non-null  object 
 5   Hidden Gem Score       13379 non-null  float64
 6   Country Availability   15461 non-null  object 
 7   Runtime                15479 non-null  object 
 8   Director               10772 non-null  object 
 9   Writer                 11150 non-null  object 
 10  Actors                 13555 non-null  object 
 11  View Rating            8456 non-null   object 
 12  IMDb Score             13381 non-null  float64
 13  Rotten Tomatoes Score  6382 non-null   float64
 14  Metacritic Score       4336 non-null   float64
 15  Aw

Here we can see, the Non-NULL and NULL 

In [39]:
# Checking out the NULL values in the given dataset
file.isnull().sum()

Title                        0
Genre                     1710
Tags                        67
Languages                 1935
Series or Movie              0
Hidden Gem Score          2101
Country Availability        19
Runtime                      1
Director                  4708
Writer                    4330
Actors                    1925
View Rating               7024
IMDb Score                2099
Rotten Tomatoes Score     9098
Metacritic Score         11144
Awards Received           9405
Awards Nominated For      7819
Boxoffice                11473
Release Date              2107
Netflix Release Date         0
Production House         10331
Netflix Link                 0
IMDb Link                 2303
Summary                      9
IMDb Votes                2101
Image                        0
Poster                    3638
TMDb Trailer              8286
Trailer Site              8286
dtype: int64

Here we can see the NULL values in eah column.

**Thus we will be dropping the columns which have NULL values and we dont need them.**
Here, most of the columns with **maximum** number of `NULL VALUES` are :
1. Metacritic Score         11144
2. Boxoffice                11473
3. Production House         10331
4. Netflix Link                 0 (we don't need this column)
5. IMDb Link                 2303

In [42]:
# Dropping these columns from the dataset
file = file.drop(columns = ['Metacritic Score', 'Boxoffice', 'Production House', 'Netflix Link', 'IMDb Link'], axis = 1)

KeyError: "['Metacritic Score' 'Boxoffice' 'Production House' 'Netflix Link'\n 'IMDb Link'] not found in axis"

In [43]:
file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15480 entries, 0 to 15479
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Title                  15480 non-null  object 
 1   Genre                  13770 non-null  object 
 2   Tags                   15413 non-null  object 
 3   Languages              13545 non-null  object 
 4   Series or Movie        15480 non-null  object 
 5   Hidden Gem Score       13379 non-null  float64
 6   Country Availability   15461 non-null  object 
 7   Runtime                15479 non-null  object 
 8   Director               10772 non-null  object 
 9   Writer                 11150 non-null  object 
 10  Actors                 13555 non-null  object 
 11  View Rating            8456 non-null   object 
 12  IMDb Score             13381 non-null  float64
 13  Rotten Tomatoes Score  6382 non-null   float64
 14  Awards Received        6075 non-null   float64
 15  Aw

As here we have dropped 5 columns from the Dataset, as they were of no use to us.

In [44]:
# Creting the ID columns for each row.
st = 1
file.insert(0, 'ID', range(st, st + file.shape[0]))

In [29]:
file.head()

ID                Title                                   Genre  \
0   1     Lets Fight Ghost  Crime, Drama, Fantasy, Horror, Romance   
1   2  HOW TO BUILD A GIRL                                  Comedy   
2   3           Centigrade                         Drama, Thriller   
3   4                ANNE+                                   Drama   
4   5                Moxie                 Animation, Short, Drama   

                                                Tags         Languages  \
0  Comedy Programmes,Romantic TV Comedies,Horror ...  Swedish, Spanish   
1       Dramas,Comedies,Films Based on Books,British           English   
2                                          Thrillers           English   
3        TV Dramas,Romantic TV Dramas,Dutch TV Shows           Turkish   
4  Social Issue Dramas,Teen Movies,Dramas,Comedie...           English   

  Series or Movie  Hidden Gem Score  \
0          Series               4.3   
1           Movie               7.0   
2           Movie               6.4   
3          Series               7.7   
4           Movie               8.1   

                                Country Availability       Runtime  \
0                                           Thailand  < 30 minutes   
1                                             Canada      1-2 hour   
2                                             Canada      1-2 hour   
3                                Belgium,Netherlands  < 30 minutes   
4  Lithuania,Poland,France,Iceland,Italy,Spain,Gr...      1-2 hour   

          Director  ... Awards Received Awards Nominated For Release Date  \
0  Tomas Alfredson  ...            74.0                 57.0  12 Dec 2008   
1    Coky Giedroyc  ...             1.0                  NaN  08 May 2020   
2    Brendan Walsh  ...             NaN                  NaN  28 Aug 2020   
3              NaN  ...             1.0                  NaN  01 Oct 2016   
4    Stephen Irwin  ...             NaN                  4.0  22 Sep 2011   

   Netflix Release Date                                            Summary  \
0            2021-03-04  A med student with a supernatural gift tries t...   
1            2021-03-04  When nerdy Johanna moves to London, things get...   
2            2021-03-04  Trapped in a frozen car during a blizzard, a p...   
3            2021-03-04  Upon moving into a new place, a 20-something r...   
4            2021-03-04  Inspired by her moms rebellious past and a con...   

   IMDb Votes                                              Image  \
0    205926.0  https://occ-0-4708-64.1.nflxso.net/dnm/api/v6/...   
1      2838.0  https://occ-0-1081-999.1.nflxso.net/dnm/api/v6...   
2      1720.0  https://occ-0-1081-999.1.nflxso.net/dnm/api/v6...   
3      1147.0  https://occ-0-1489-1490.1.nflxso.net/dnm/api/v...   
4        63.0  https://occ-0-4039-1500.1.nflxso.net/dnm/api/v...   

                                              Poster  \
0  https://m.media-amazon.com/images/M/MV5BOWM4NT...   
1  https://m.media-amazon.com/images/M/MV5BZGUyN2...   
2  https://m.media-amazon.com/images/M/MV5BODM2MD...   
3  https://m.media-amazon.com/images/M/MV5BNWRkMz...   
4  https://m.media-amazon.com/images/M/MV5BODYyNW...   

                                  TMDb Trailer Trailer Site  
0                                          NaN          NaN  
1  https://www.youtube.com/watch?v=eIbcxPy4okQ      YouTube  
2  https://www.youtube.com/watch?v=0RvV7TNUlkQ      YouTube  
3                                          NaN          NaN  
4                                          NaN          NaN  

[5 rows x 25 columns]

In [45]:
# Deleting the NULL values from the dataset.
file.dropna(axis=0, how='any')

ID                                     Title  \
15        16                                     Joker   
16        17                                         I   
17        18                          Harrys Daughters   
28        29                                The Closet   
60        61                           Ordinary People   
...      ...                                       ...   
15459  15460                                One Chance   
15460  15461  The Twilight Saga: Breaking Dawn: Part 1   
15461  15462                         One for the Money   
15462  15463                                   4.3.2.1   
15463  15464                                        13   

                                              Genre  \
15                           Crime, Drama, Thriller   
16               Action, Adventure, Fantasy, Sci-Fi   
17               Adventure, Drama, Fantasy, Mystery   
28                                           Comedy   
60                                            Drama   
...                                             ...   
15459               Biography, Comedy, Drama, Music   
15460  Adventure, Drama, Fantasy, Romance, Thriller   
15461               Action, Comedy, Crime, Thriller   
15462                               Crime, Thriller   
15463                        Crime, Drama, Thriller   

                                                    Tags            Languages  \
15     Dark Comedies,Crime Comedies,Dramas,Comedies,C...              English   
16                                 Dramas,Swedish Movies    English, Sanskrit   
17                                 Dramas,Swedish Movies              English   
28                 Korean Movies,Horror Movies,Mysteries               French   
60         Korean Movies,Thriller Movies,Crime Thrillers              English   
...                                                  ...                  ...   
15459  Dramas,Biographical Dramas,Dramas based on rea...     English, Italian   
15460  Dramas,Romantic Dramas,Dramas based on a book,...  English, Portuguese   
15461  Romantic Comedies,Action Comedies,Comedies,Pol...              English   
15462  Thrillers,Steamy Thrillers,Police Thrillers,Po...              English   
15463  Thrillers,Gangster Films,Police Thrillers,Poli...              English   

      Series or Movie  Hidden Gem Score  \
15              Movie               3.5   
16              Movie               2.8   
17              Movie               4.4   
28              Movie               3.8   
60              Movie               4.2   
...               ...               ...   
15459           Movie               3.0   
15460           Movie               2.0   
15461           Movie               1.3   
15462           Movie               2.4   
15463           Movie               1.6   

                                    Country Availability   Runtime  \
15     Lithuania,Poland,France,Italy,Spain,Greece,Bel...  1-2 hour   
16     Lithuania,Poland,France,Italy,Spain,Greece,Cze...  1-2 hour   
17     Lithuania,Poland,France,Italy,Spain,Greece,Cze...  1-2 hour   
28                                           South Korea  1-2 hour   
60                                           South Korea  1-2 hour   
...                                                  ...       ...   
15459                                     Japan,Malaysia  1-2 hour   
15460  Canada,Romania,Japan,Switzerland,United Kingdo...   > 2 hrs   
15461                                             Canada  1-2 hour   
15462                                              India  1-2 hour   
15463            Romania,Hungary,Czech Republic,Slovakia  1-2 hour   

                      Director  ... Awards Received Awards Nominated For  \
15               Todd Phillips  ...           112.0                228.0   
16                George Lucas  ...            26.0                 69.0   
17                 David Yates  ...            46.0                 94.0   
28               

In [46]:
file.head(2)

ID                Title                                   Genre  \
0   1     Lets Fight Ghost  Crime, Drama, Fantasy, Horror, Romance   
1   2  HOW TO BUILD A GIRL                                  Comedy   

                                                Tags         Languages  \
0  Comedy Programmes,Romantic TV Comedies,Horror ...  Swedish, Spanish   
1       Dramas,Comedies,Films Based on Books,British           English   

  Series or Movie  Hidden Gem Score Country Availability       Runtime  \
0          Series               4.3             Thailand  < 30 minutes   
1           Movie               7.0               Canada      1-2 hour   

          Director  ... Awards Received Awards Nominated For Release Date  \
0  Tomas Alfredson  ...            74.0                 57.0  12 Dec 2008   
1    Coky Giedroyc  ...             1.0                  NaN  08 May 2020   

   Netflix Release Date                                            Summary  \
0            2021-03-04  A med student with a supernatural gift tries t...   
1            2021-03-04  When nerdy Johanna moves to London, things get...   

   IMDb Votes                                              Image  \
0    205926.0  https://occ-0-4708-64.1.nflxso.net/dnm/api/v6/...   
1      2838.0  https://occ-0-1081-999.1.nflxso.net/dnm/api/v6...   

                                              Poster  \
0  https://m.media-amazon.com/images/M/MV5BOWM4NT...   
1  https://m.media-amazon.com/images/M/MV5BZGUyN2...   

                                  TMDb Trailer Trailer Site  
0                                          NaN          NaN  
1  https://www.youtube.com/watch?v=eIbcxPy4okQ      YouTube  

[2 rows x 25 columns]